src/test/java/io/kestra/plugin/mongodb/AggregateTest.java

```diff
@@ -0,0 +1,327 @@
+package io.kestra.plugin.mongodb;
+
+import com.google.common.collect.ImmutableMap;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.runners.RunContextFactory;
+import io.kestra.core.serializers.FileSerde;
+import io.kestra.core.serializers.JacksonMapper;
+import io.kestra.core.junit.annotations.KestraTest;
+import io.kestra.core.utils.IdUtils;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.*;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+@KestraTest
+class AggregateTest {
+    @Inject
+    private RunContextFactory runContextFactory;
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void run() throws Exception {
+        RunContext runContext = runContextFactory.of(ImmutableMap.of());
+
+        Aggregate aggregate = Aggregate.builder()
+            .connection(MongoDbConnection.builder()
+                .uri(Property.ofValue("mongodb://root:example@localhost:27017/?authSource=admin"))
```

Comment:
The test cannot work without a MongoDB run thanks to testcontainers here or using a mock.

Reply:
went for TestContainers, to be confident of the behavior with a real mongodb instance